docs/migration/cosmoshub-4-vega-upgrade.md

```diff
@@ -1,4 +1,217 @@
+
+### Current runtime, cosmoshub-4 (pre-Vega upgrade) is running Gaia v5.0.0
+
+The Cosmos Hub mainnet network, `cosmoshub-4`, is currently running [Gaia v5.0.0](https://github.com/cosmos/gaia/releases/tag/v5.0.0). We anticipate that operators who are running earlier versions of Gaia, e.g., v5.0.x, will be able to upgrade successfully; however, this is untested and it is up to operators to ensure that their systems are capable of performing the upgrade.
+
+### Target runtime, cosmoshub-4 (post-Vega upgrade) will run Gaia v6.0.0
+
+The Comsos Hub mainnet network, `cosmoshub-4`, will run [Gaia v6.0.0](https://github.com/cosmos/gaia/releases/tag/v6.0.0). Operators _MUST_ use this version post-upgrade to remain connected to the network.
+
+## Vega upgrade steps
+There are 2 major ways to upgrade a node: 
+- manual upgrade
+- upgrade by [cosmovisor](https://github.com/cosmos/cosmos-sdk/tree/master/cosmovisor)
+    - with manually prepare new binary
+    - with manually prepare new binary
+If you prefer to use cosmovisor to upgrade, a preparation work is needed before upgrade.
+
+### method I: manual upgrade
+Run Gaia v5.0.x till upgrade height, the node will panic:
+```shell
+ERR UPGRADE "Vega" NEEDED at height: 8695000
+
+panic: UPGRADE "Vega" NEEDED at height: 8695000
+```
+Stop the node, and install Gaia v6.0.0 and re-start by `gaiad start`.
+
+It may take 20 min to a few hours till validators with total sum power > 2/3 complete their nodes upgrades and by that moment, the chain can continue to produce blocks.
+
+### Method II: upgrade by cosmovisor with manually prepare the Gaia v6.0.0 binary
+#### Preparation
+
+Install cosmovisor latest version:
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
+```
+
+Create a cosmovisor folder:
+
+create cosmovisor folder inside gaia home and move Gaia v5.0.0 into ` $GAIA_HOME/cosmovisor/genesis/bin`
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+```
+build Gaia v6.0.0, and move gaiad v6.0.0 to `$GAIA_HOME/cosmovisor/upgrades/Vega/bin`
+
+```shell
+mkdir -p  $GAIA_HOME/cosmovisor/upgrades/Vega/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/upgrades/Vega/bin
+```
+Then you should get the following structure:
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiadv5.0.0
+└── upgrades
+└── Vega
+└── bin
+    └── gaiadv6.0.0
+```
+Export the environmental variables:
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME= $GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node: 
+```shell
+cosmovisor start --x-crisis-skip-assert-invariants
+```
+ ????
```

Comment:
```suggestion
Skipping the invariant checks is strongly encouraged since it decreases the upgrade time significantly and since there are some other improvements coming to the crisis module in the next release of the Cosmos SDK.
```